LanguageFeatures/Augmentation-libraries/augmented_expression_A02_t06.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting setters: Within an augmenting setter `augmented` must be
+///   followed by an `=` and will directly invoke the augmented setter. If
+///   augmenting a field with a setter, this will invoke the implicit setter
+///   from the augmented field.
+///
+/// @description Checks that within an augmented setter `augmented=` can be used
+/// to invoke appropriate setter in the current scope.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'augmented_expression_A02_t06_lib.dart';
+import '../../Utils/expect.dart';
+
+String _log = "";
+
+void set augmented(String value) {
+  _log = "Setter augmented: $value";
+}
+
+void set topLevelSetter(String value) {
+  augmented = value;
```

Comment:
Perhaps?:
```suggestion
  augmented = '$value, via original declaration';
```
in order to detect that the invocations performed in the augmentations will call this declaration, they will not call the top-level setter named `augmented` directly.